### ISSUE TITLE
Preserve path option

### ISSUE THREAD

**Issue by dskaggs** [tfeldmann/organize#9]
It would be very helpful to have an option to replicate the file path from root of the source folder to the destination folder on a move or copy action. I subscribe to a system that downloads statements from over a dozen companies that I do business with and files them into a sub-folder for each account. I'd like to have an automated process run periodically to file those into a folder of the same name, but under the current year so that, at year's end, all the documents for that particular year are filed together

For instance, given the following input file:
`/source-folder/my-bank-name/monthly-statement-july.pdf`

move to the following destination file:
`/destination-folder/2018/my-bank-name/monthly-statement-july.pdf`

One idea might be to have a rule similar to the following:
```
rules:
  - folders: '/source-folder'
    subfolders: true
    filters:
      - Extension:
          - pdf
          - docx
      - LastModified
    actions:
      - Move: '/destination-folder/{lastmodified.year}/'
          - preservepath: true
```
Another might be even simpler:
```
rules:
  - folders: '/source-folder'
    subfolders: true
    filters:
      - Extension:
          - pdf
          - docx
      - LastModified
    actions:
      - Move: '/destination-folder/{lastmodified.year}/{path}/'
```

**Comment by dskaggs** [comment]
Awesome! I'm working on a Docker image for this so I can run it on my server at home and have it executed periodically via cron (https://hub.docker.com/r/dskaggs/docker-organize/). Thanks a bunch!

**Comment by tfeldmann** [comment]
Very cool 👍 Thank you for sharing!

**Comment by dskaggs** [comment]
Just to circle back around to this, I finally got a chance to test out the `{relative_path}` feature today and found that it didn't do what I expected. In my testing today, it actually added an additional, unwanted folder to the path.


For instance, given the following input file:
`/source-folder/my-bank-name/monthly-statement-july.pdf`

Using the following action:
`- Move: '/destination-folder/{lastmodified.year}/{relative_path}'`

Results in a destination file of:
`/destination-folder/2018/my-bank-name/monthly-statement-july.pdf/monthly-statement-july.pdf`

As a result, I had to change the rule to be:
`- Move: '/destination-folder/{lastmodified.year}/{path.parent.name}/`
in order to remove the extra folder.

Not sure if that's a bug or just the way my files were laid out, but the above edited rule worked.
